lib/internal/bootstrap_node.js

```diff
@@ -557,9 +571,37 @@
     nativeModule.cache();
     nativeModule.compile();
 
+    if (config.experimentalModules) {
+      const ns = NativeModule._nsCache[id] = Object.create(null);
+      if (typeof nativeModule.exports === 'function')
+        ns.default = nativeModule.exports;
+
+      const descriptors =
+        Object.getOwnPropertyDescriptors(nativeModule.exports);
+      for (const [name, d] of Object.entries(descriptors)) {
+        if (!d.enumerable || d.set !== undefined)
+          continue;
```

Comment:
Perhaps let's leave setters in for now. Things like `process.domain` could be updated so that their setters update the live bindings on the ES module here, if we make the cached object here a dynamic module object to support that.

Reply:
Oh cool. So of the remaining filters do we know which non-enumerable or setter properties are being excluded?

Reply:
`import { prototype } from 'events'`, `crypto.fips = true`, etc

Reply:
Do we have a complete list of what each are (non-enumerable, setter, both)?

Reply:
| NON-ENUMERABLE | GET | SET | NAME | NOTE |
|----------------|-----|-----|------|------|
| ✔️ |   |   | `assert.length` | |
| ✔️ |   |   | `assert.name` | |
| ✔️ |   |   | `assert.prototype` | |
| ✔️ | ✔️ | ✔️ | `crypto.fips` | bean'd |
|   | ✔️ | ✔️ | `crypto.DEFAULT_ENCODING` | dep map, bean'd |
|   | ✔️ |   | `crypto.createCredentials` | dep map |
|   | ✔️ |   | `crypto.Credentials` | dep map |
| ✔️ |   |   | `events.length` | |
| ✔️ |   |   | `events.name` | |
| ✔️ |   |   | `events.prototype` | |
|   | ✔️ | ✔️ | `events.defaultMaxListeners` | on default export |
| ✔️ | ✔️ | ✔️ | `fs.SyncWriteStream` | |
| ✔️ |   |   | `stream.length` | |
| ✔️ |   |   | `stream.name` | |
| ✔️ |   |   | `stream.prototype` | |
| ✔️ |   |   | `process._promiseRejectEvent` | default export |
|   | ✔️ | ✔️ | `process.domain` | |
|   | ✔️ |   | `process.stdout` | |
|   | ✔️ |   | `process.stderr` | |
|   | ✔️ |   | `process.stdin` | |

Reply:
Cool. FWIW in my implementation I ended up incidentally filtering non-enumerable because I just used an `Object.keys` equiv to get the values. I support getters, though I don't have live bindings supported in the cjs projection of builtins at the moment.

If there's no technical hurdle might as well support non-enumerable though.